go.mod

```diff
@@ -9,62 +9,76 @@ go 1.16
-	github.com/onsi/gomega v1.10.1
+	github.com/onsi/ginkgo v1.16.1
+	github.com/onsi/gomega v1.11.0
 	github.com/pavel-v-chernykh/keystore-go v2.1.0+incompatible
 	github.com/pkg/errors v0.9.1
-	github.com/prometheus/client_golang v1.7.1
+	github.com/prometheus/client_golang v1.9.0
 	github.com/sergi/go-diff v1.1.0
 	github.com/smartystreets/assertions v1.2.0 // indirect
-	github.com/spf13/cobra v1.0.0
+	github.com/spf13/cobra v1.1.3
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.6.1
-	golang.org/x/crypto v0.0.0-20200622213623-75b288015ac9
-	golang.org/x/net v0.0.0-20210226172049-e18ecbb05110
+	golang.org/x/crypto v0.0.0-20210220033148-5ea612d1eb83
+	golang.org/x/net v0.0.0-20210405180319-a5a99cb37ef4
 	golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
-	golang.org/x/sync v0.0.0-20190911185100-cd5d95a43a6e
-	google.golang.org/api v0.15.0
+	golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
+	google.golang.org/api v0.20.0
+	google.golang.org/grpc v1.35.0 // indirect
 	gopkg.in/ini.v1 v1.52.0 // indirect
 	gopkg.in/yaml.v2 v2.4.0
-	gopkg.in/yaml.v3 v3.0.0-20200605160147-a5ece683394c // indirect
-	k8s.io/api v0.19.0
-	k8s.io/apiextensions-apiserver v0.19.0
-	k8s.io/apimachinery v0.19.0
-	k8s.io/apiserver v0.19.0
-	k8s.io/cli-runtime v0.19.0
-	k8s.io/client-go v0.19.0
-	k8s.io/code-generator v0.19.0
-	k8s.io/component-base v0.19.0
-	k8s.io/klog/v2 v2.3.0
-	k8s.io/kube-aggregator v0.19.0
-	k8s.io/kube-openapi v0.0.0-20200805222855-6aeccd4b50c6
-	k8s.io/kubectl v0.19.0
-	k8s.io/utils v0.0.0-20200729134348-d5654de09c73
-	sigs.k8s.io/controller-runtime v0.6.2
-	sigs.k8s.io/controller-tools v0.2.9-0.20200414181213-645d44dca7c0
+	k8s.io/api v0.21.0
+	k8s.io/apiextensions-apiserver v0.21.0
+	k8s.io/apimachinery v0.21.0
+	k8s.io/apiserver v0.21.0
+	k8s.io/cli-runtime v0.21.0
+	k8s.io/client-go v0.21.0
+	k8s.io/code-generator v0.21.0
+	k8s.io/component-base v0.21.0
+	k8s.io/klog/v2 v2.8.0
+	k8s.io/kube-aggregator v0.21.0
+	k8s.io/kube-openapi v0.0.0-20210305001622-591a79e4bda7
+	k8s.io/kubectl v0.21.0
+	k8s.io/utils v0.0.0-20210111153108-fddb29f9d009
+	sigs.k8s.io/controller-runtime v0.9.0-beta.1
+	sigs.k8s.io/controller-tools v0.6.0-beta.0
 	sigs.k8s.io/testing_frameworks v0.1.2
 	sigs.k8s.io/yaml v1.2.0
 	software.sslmate.com/src/go-pkcs12 v0.0.0-20200830195227-52f69702a001
 )
+
+replace golang.org/x/net => golang.org/x/net v0.0.0-20210224082022-3d97a244fca7
+
+replace github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.4.1
+
+replace github.com/onsi/ginkgo => github.com/onsi/ginkgo v1.12.1
+
+replace github.com/onsi/gomega => github.com/onsi/gomega v1.10.1
+
+replace google.golang.org/grpc => google.golang.org/grpc v1.27.1
+
+replace k8s.io/code-generator => github.com/kmodules/code-generator v0.21.1-rc.0.0.20210428003838-7eafae069eb0
+
+replace k8s.io/gengo => github.com/kmodules/gengo v0.0.0-20210428002657-a8850da697c2
+
+replace k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20210305001622-591a79e4bda7
```

Comment:
Sorry for the late review (non-blocking). I think we can reduce the number of replaces to:

```
diff --git a/go.mod b/go.mod
index 3e97e116b..71692e0a3 100644
--- a/go.mod
+++ b/go.mod
@@ -44,9 +44,8 @@ require (
        golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
        golang.org/x/sync v0.0.0-20201020160332-67f06af15bc9
        google.golang.org/api v0.20.0
-       google.golang.org/grpc v1.35.0 // indirect
        gopkg.in/ini.v1 v1.52.0 // indirect
-       gopkg.in/yaml.v2 v2.4.0
+       gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
        k8s.io/api v0.21.0
        k8s.io/apiextensions-apiserver v0.21.0
        k8s.io/apimachinery v0.21.0
@@ -68,15 +67,3 @@ require (
 )
 
 replace github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.4.1
-
-replace github.com/onsi/ginkgo => github.com/onsi/ginkgo v1.12.1
-
-replace github.com/onsi/gomega => github.com/onsi/gomega v1.10.1
-
-replace google.golang.org/grpc => google.golang.org/grpc v1.27.1
-
-replace k8s.io/code-generator => github.com/kmodules/code-generator v0.21.1-rc.0.0.20210428003838-7eafae069eb0
-
-replace k8s.io/gengo => github.com/kmodules/gengo v0.0.0-20210428002657-a8850da697c2
-
-replace k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20210305001622-591a79e4bda7
```

```
diff --git a/hack/filter-crd/main.go b/hack/filter-crd/main.go
index 8a2f35d35..c671a8797 100644
--- a/hack/filter-crd/main.go
+++ b/hack/filter-crd/main.go
@@ -23,7 +23,7 @@ import (
        "os"
        "strings"
 
-       "gopkg.in/yaml.v2"
+       "gopkg.in/yaml.v3"
 )
```